pkg/kubectl/cmd/convert.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/kubernetes/pkg/api/latest"
+	"k8s.io/kubernetes/pkg/api/registered"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	convert_long = `Convert config files between different API versions. Both YAML
+and JSON formats are accepted.
+
+The command takes filename, directory, or URL as input, and convert it into format
+of version specified by -output-version flag. If target version is not specified or
+not supported, convert to latest version. 
+
+The default output will be printed to stdout in YAML format. One can use -o option 
+to change to output destination.
+`
+	convert_example = `// Convert 'pod.yaml' to latest version and print to stdout.
+$ kubectl convert -f pod.yaml
+
+// Convert 'pod.yaml' to version v1 and output to 'b3pod.yaml'.
+$ kubectl convert -f pod.yaml --output-version v1 -t b3pod.yaml
+
+// Convert 'pod.yaml' to latest version locally and print to stdout in json format.
+$ kubectl convert -f pod.yaml --local -o json
+
+// Convert all files under current directory to latest version and create them all.
+$ kubectl convert -f . | kubectl create -f -
+`
+)
+
+// NewCmdConvert creates a command object for the generic "convert" action, which
+// translates the config file into a given version.
+func NewCmdConvert(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	options := &ConvertOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "convert -f FILENAME",
+		Short:   "Convert config files between different API versions",
+		Long:    convert_long,
+		Example: convert_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := options.Complete(f, out, cmd, args)
+			cmdutil.CheckErr(err)
+			err = options.RunConvert()
```

Comment:
It works for directory. 
In my mind `get` should retrieve live status from server, `--local` doesn't mean much for `get`. If I want to "get" resources from a local file, I can just open the file. 
And for `get`, it's better to keep the information unchanged. So I suggest not do convertion in `get` command to avoid confusion.